### PR TITLE
Update Core Maps to beta.17

### PR DESCRIPTION
**Pull request by julianrex**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [X] Briefly describe the changes in this PR.
 - [ ] ~Include before/after visuals or gifs if this PR includes visual changes.~
 - [ ] ~Write tests for all new functionality. If tests were not written, please explain why.~
 - [X] Add example if relevant. _Examples updated_.
 - [X] Document any changes to public APIs.
 - [X] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [X] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>Updates dependencies to MapboxCoreMaps 10.0.0-beta.17 and MapboxCommon 10.0.0.</changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This PR updates the MapboxMaps SDK to use [MapboxCoreMaps 10.0.0-beta.17](https://github.com/mapbox/mapbox-core-maps-ios/releases/tag/v10.0.0-beta.17) and removes `MapboxCommon` as an explicit dependency (MapboxCommon is bumped to [10.0.2](https://github.com/mapbox/mapbox-common-ios/releases/tag/v10.0.2) implicitly); there are several changes that affect developers using the MapboxMaps SDK:
 
⚠️ **Breaking changes**

Map events have been renamed and aligned between Android, iOS and GL-JS:

- Renamed `mapLoadingFinished` to `mapLoaded` 
- Added `sourceDataLoaded` and `styleDataLoaded`; aligns events with GL-JS events ([map.event:styledata](https://docs.mapbox.com/mapbox-gl-js/api/map/#map.event:styledata), [map.event:sourcedata](https://docs.mapbox.com/mapbox-gl-js/api/map/#map.event:sourcedata))
- Rename `styleFullyLoaded` to `styleLoaded`; renamed for consistency with `mapLoaded`, it means the map's resources are fully loaded. Partial resource loading events are exposed through `styleDataLoaded`, `sourceDataLoaded`
- Begin/end timestamps of the associated event have moved to the `data` property. `end` is optional.
- `mapLoadingError` now provides information about which kind of resource failed to load (as part of the `data` property)

**Additional improvements**

- Fixes a crash with `OnlineFileSource` destruction, which was occurring when running the `Examples` app as tests.
- Fixes a bug with fill extrusion querying when terrain is enabled.
- Improves render performance when terrain is enabled.

### User impact

The biggest change is the event name refactoring. Please see the migration guide and the event lifecycle diagram below for more information.
<details>
<summary>
Simplified diagram for events emitted by the Map object.
</summary>
<pre>
 *
 * ┌─────────────┐               ┌─────────┐                   ┌──────────────┐
 * │ Application │               │   Map   │                   │ResourceLoader│
 * └──────┬──────┘               └────┬────┘                   └───────┬──────┘
 *        │                           │                                │
 *        ├───────setStyleURI────────▶│                                │
 *        │                           ├───────────get style───────────▶│
 *        │                           │                                │
 *        │                           │◀─────────style data────────────┤
 *        │                           │                                │
 *        │                           ├─parse style─┐                  │
 *        │                           │             │                  │
 *        │      StyleDataLoaded      ◀─────────────┘                  │
 *        │◀────{"type": "style"}─────┤                                │
 *        │                           ├─────────get sprite────────────▶│
 *        │                           │                                │
 *        │                           │◀────────sprite data────────────┤
 *        │                           │                                │
 *        │                           ├──────parse sprite───────┐      │
 *        │                           │                         │      │
 *        │      StyleDataLoaded      ◀─────────────────────────┘      │
 *        │◀───{"type": "sprite"}─────┤                                │
 *        │                           ├─────get source TileJSON(s)────▶│
 *        │                           │                                │
 *        │     SourceDataLoaded      │◀─────parse TileJSON data───────┤
 *        │◀──{"type": "metadata"}────┤                                │
 *        │                           │                                │
 *        │                           │                                │
 *        │      StyleDataLoaded      │                                │
 *        │◀───{"type": "sources"}────┤                                │
 *        │                           ├──────────get tiles────────────▶│
 *        │                           │                                │
 *        │◀───────StyleLoaded────────┤                                │
 *        │                           │                                │
 *        │     SourceDataLoaded      │◀─────────tile data─────────────┤
 *        │◀────{"type": "tile"}──────┤                                │
 *        │                           │                                │
 *        │                           │                                │
 *        │◀────RenderFrameStarted────┤                                │
 *        │                           ├─────render─────┐               │
 *        │                           │                │               │
 *        │                           ◀────────────────┘               │
 *        │◀───RenderFrameFinished────┤                                │
 *        │                           ├──render, all tiles loaded──┐   │
 *        │                           │                            │   │
 *        │                           ◀────────────────────────────┘   │
 *        │◀────────MapLoaded─────────┤                                │
 *        │                           │                                │
 *        │                           │                                │
 *        │◀─────────MapIdle──────────┤                                │
 *        │                    ┌ ─── ─┴─ ─── ┐                         │
 *        │                    │   offline   │                         │
 *        │                    └ ─── ─┬─ ─── ┘                         │
 *        │                           │                                │
 *        ├─────────setCamera────────▶│                                │
 *        │                           ├───────────get tiles───────────▶│
 *        │                           │                                │
 *        │                           │┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─   │
 *        │◀─────────MapIdle──────────┤   waiting for connectivity  │  │
 *        │                           ││  Map renders cached data      │
 *        │                           │ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘  │
 *        │                           │                                │
 *
</pre>
</details>

### CODE REVIEW

File: Tests/MapboxMapsTests/Style/Generated/IntegrationTests/Layers/BackgroundLayerIntegrationTests.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ class BackgroundLayerIntegrationTests: MapViewIntegrationTestCase {
         style.styleURL = .streets
 
         didFinishLoadingStyle = { _ in
-
+print("hello")
```
Comment (nishant-karajgikar) [inline Tests/MapboxMapsTests/Style/Generated/IntegrationTests/Layers/BackgroundLayerIntegrationTests.swift:31]: mistake?

Comment (julianrex) [reply Tests/MapboxMapsTests/Style/Generated/IntegrationTests/Layers/BackgroundLayerIntegrationTests.swift:31]: haha!

**Review by nishant-karajgikar — APPROVED** [review]

File: Apps/Examples/Examples/Assets.xcassets/AccentColor.colorset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (nishant-karajgikar) [inline Apps/Examples/Examples/Assets.xcassets/AccentColor.colorset/Contents.json:1]: Why is this getting added here?

Comment (julianrex) [reply Apps/Examples/Examples/Assets.xcassets/AccentColor.colorset/Contents.json:1]: Gets rid of a build warning about `AccentColor`.

File: Examples/Examples/Assets.xcassets/AccentColor.colorset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (macdrevx) [inline Examples/Examples/Assets.xcassets/AccentColor.colorset/Contents.json:1]: This file looks redundant with https://github.com/mapbox/mapbox-maps-ios/pull/193/files#diff-02029dab4ca42e1076eb249084403aac9d416ba002e7bcb053b856d94c51bd26R1

Comment (julianrex) [reply Examples/Examples/Assets.xcassets/AccentColor.colorset/Contents.json]: Removed.

**Pull request merged**